index.js

```diff
@@ -35,23 +35,28 @@ module.exports = function (src, dest, opts, cb) {
 				return;
 			}
 
-			var read = fs.createReadStream(src);
-			var write = fs.createWriteStream(dest);
-
-			read.on('error', done);
-			write.on('error', done);
-			write.on('close', function () {
-				fs.lstat(src, function (err, stats) {
-					if (err) {
-						done(err);
-						return;
-					}
-
+			fs.lstat(src, function (err, stats) {
+				if (err) {
+					done(err);
+					return;
+				}
+
+				if (stats.isDirectory()) {
```

Comment:
@schnittstabil Is right about race condition with stat. Basically you should never use the result of fs.stat/exists for any following fs read/write operation as the state of truth might change between the calls.